File: programmers/js/basic/181930.js
// 주사위 게임 2
function solution(a, b, c) {
  let answer = a + b + c;

  if (a !== b && b !== c && c !== a) {
    return answer;
  }

  answer *= a ** 2 + b ** 2 + c ** 2;

  if ((a === b && b !== c) || (b === c && c !== a) || (c === a && a !== b)) {
    return answer;
  }

  answer *= a ** 3 + b ** 3 + c ** 3;

  if (a === b && b === c) {
    return answer;
  }
}

File: programmers/java/basic/181930.java
// 주사위 게임 2

class Solution {
    public int calcPowSum(int a, int b, int c, int powNumber) {
        return (int) Math.pow(a, powNumber) + (int) Math.pow(b, powNumber) + (int) Math.pow(c, powNumber);
    }
    
    public int solution(int a, int b, int c) {
        int score = a + b + c;
        
        if (a == b && b == c) {
            score *= calcPowSum(a, b, c, 2) * calcPowSum(a, b, c, 3);
        } else if ((a == b && b != c) || (a == c && c != b) || (b == c && c != a)) {
            score *= calcPowSum(a, b, c, 2);
        }
        
        return score;
    }
}